source/Vecerdi.Logging/UnityLoggerProvider.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Logging;
+
+namespace Vecerdi.Logging.Unity {
+    /// <summary>
+    /// Logger provider that routes Microsoft.Extensions.Logging calls to Unity's Debug.Log* methods.
+    /// This class is designed to be used in environments where both Microsoft.Extensions.Logging 
+    /// and Unity are available.
+    /// </summary>
+    public class UnityLoggerProvider : ILoggerProvider {
```

Comment:
Please move Unity-specific APIs into the Unity integration so we can directly reference Unity APIs and not have to rely on reflection